libraries/browser-tracker-core/src/tracker/index.ts

```diff
@@ -0,0 +1,1247 @@
+     * no-op if anonymousTracking enabled, will still set cookies if anonymousSessionTracking is enabled
+     * Sets a cookie based on the storage strategy:
+     * - if 'localStorage': attemps to write to local storage
+     * - if 'cookie' or 'cookieAndLocalStorage': writes to cookies
+     * - otherwise: no-op
+     */
+    function setCookie(name: string, value: string, timeout: number) {
+      if (configAnonymousTracking && !configAnonymousSessionTracking) {
+        return;
+      }
+
+      if (configStateStorageStrategy == 'localStorage') {
+        attemptWriteLocalStorage(name, value, timeout);
+      } else if (configStateStorageStrategy == 'cookie' || configStateStorageStrategy == 'cookieAndLocalStorage') {
+        cookie(name, value, timeout, configCookiePath, configCookieDomain, configCookieSameSite, configCookieSecure);
+      }
+    }
+
+    /**
+     * Generate a pseudo-unique ID to identify this user
+     */
+    function createNewDomainUserId() {
+      return uuid();
+    }
+
+    /**
+     * Clears all cookie and local storage for id and ses values
+     */
+    function deleteCookies() {
+      const idname = getSnowplowCookieName('id');
+      const sesname = getSnowplowCookieName('ses');
+      attemptDeleteLocalStorage(idname);
+      attemptDeleteLocalStorage(sesname);
+      deleteCookie(idname, configCookieDomain, configCookieSameSite, configCookieSecure);
+      deleteCookie(sesname, configCookieDomain, configCookieSameSite, configCookieSecure);
+    }
+
+    /*
+     * Load the domain user ID and the session ID
+     * Set the cookies (if cookies are enabled)
+     */
+    function initializeIdsAndCookies() {
+      if (configAnonymousTracking && !configAnonymousSessionTracking) {
+        return;
+      }
+
+      var sesCookieSet = configStateStorageStrategy != 'none' && !!getSnowplowCookieValue('ses');
+      var idCookieComponents = loadDomainUserIdCookie();
+
+      if (idCookieComponents[1]) {
+        domainUserId = idCookieComponents[1] as string;
+      } else if (!configAnonymousTracking) {
+        domainUserId = createNewDomainUserId();
+        idCookieComponents[1] = domainUserId;
+      } else {
+        domainUserId = '';
+        idCookieComponents[1] = domainUserId;
+      }
+
+      memorizedSessionId = idCookieComponents[6] as string;
+
+      if (!sesCookieSet) {
+        // Increment the session ID
+        (idCookieComponents[3] as number)++;
+        // Create a new sessionId
+        memorizedSessionId = uuid();
+        idCookieComponents[6] = memorizedSessionId;
+        // Set lastVisitTs to currentVisitTs
+        idCookieComponents[5] = idCookieComponents[4];
+      }
+
+      if (configStateStorageStrategy != 'none') {
+        setSessionCookie();
+        // Update currentVisitTs
+        idCookieComponents[4] = Math.round(new Date().getTime() / 1000);
+        idCookieComponents.shift();
+        setDomainUserIdCookie.apply(null, idCookieComponents as any); // TODO: Remove any
```

Comment:
Is this comment left in intentionally?

Reply:
I did this because adding types to all the cookie logic was rather, uhm, tricky. I might have another go at it now you've reminded me 😉

Reply:
Cool cool just double checking it's not an oversight. :)